aws-usage-report/aws-usage-report.qmd

```diff
@@ -0,0 +1,356 @@
+ -->
+
+### Introduction
+
+A key objective of NASA Openscapes is to minimize “the time to science” for researchers. Cloud infrastructure can facilitate shortening this time. We use a 2i2c-managed JupyterHub, which lets us work in the cloud next to NASA Earthdata in AWS US-West-2. The purpose of the JupyterHub is to provide initial, exploratory experiences accessing NASA Earthdata in the cloud. It is not meant to be a long-term solution to support on-going science work or software development. For those users that decide working in the Cloud is advantageous and want to move there, we support a migration from the Hub to their own environment through Coiled.io, and are working on other "fledging" pathways.
+
+The main costs of running the JupyterHub come from two sources:
+
+1. Compute, using AWS EC2
+2. Storage using AWS EFS, via storage in users' home directories
+
+Compute costs scale up and down as the Hub is used, however storage costs are 
+fixed - we pay for "data at rest", with ongoing daily costs/GB even while the Hub is
+not running.
+
+Storing large amounts of data in the cloud can incur significant ongoing costs if not done optimally. We are developing [technical strategies and policies](https://nasa-openscapes.github.io/earthdata-cloud-cookbook/policies-admin/data-policies.html) to reduce storage costs that will keep the Openscapes 2i2c Hub a shared resource for us all to use, while also providing reusable strategies for other admins.
+
+This report is intended to give a monthly summary of usage of the hub and its 
+resources, by tracking metrics on costs and usage of key components of storage (EFS)
+and compute (EC2).
+
+```{r setup}
+#| include: false
+
+library(dplyr)
+library(kyber)
+library(ggplot2)
+library(forcats)
+library(lubridate)
+library(paws)
+library(here)
+
+knitr::opts_chunk$set(
+  echo = FALSE,
+  message = FALSE,
+  warning = FALSE
+)
+
+source(here("aws-usage-report/R/prometheus-utils.R"))
+source(here("aws-usage-report/R/aws-ce-utils.R"))
+
+if (interactive()) {
+  params <- list(year_month = format(Sys.Date() %m-% months(1), "%Y-%m"))
+}
+
+start_date <- ym(params$year_month)
+end_date <- ceiling_date(start_date, unit = "month") - days(1)
+
+cost_explorer <- paws::costexplorer()
+
+theme_set(theme_classic())
+```
+
+## Month over month changes
+
+### Total Costs
+
+```{r total-costs}
+
+# https://www.paws-r-sdk.com/docs/costexplorer_get_cost_and_usage/
+# https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_GetDimensionValues.html
+total_monthly_usage_costs <- cost_explorer$get_cost_and_usage(
+  TimePeriod = list(Start = ceiling_date(end_date %m-% months(6), unit = "month"), End = end_date),
+  Granularity = "MONTHLY",
+  Filter = list(Dimensions = list(
+    Key = "RECORD_TYPE",
+    Values = "Usage"
+  )),
+  Metrics = "UnblendedCost"
+) |>
+  ce_to_df()
+
+ggplot(total_monthly_usage_costs) +
+  geom_line(aes(x = start_date, y = UnblendedCost)) +
```

Comment:
Could you please add the text of June's cost, either via the idea of ggrepel or in the title itself? Whatever you prefer